Dockerfile

```diff
@@ -82,21 +94,23 @@ RUN --mount=type=cache,sharing=locked,target=${JULIA_DEPOT_CACHE},uid=${UID},gid
 
 # Copy the shared ephemeral Julia depot into the image and remove any installed packages
 # not used by our Manifest.toml.
-RUN --mount=type=cache,target=${JULIA_DEPOT_CACHE},uid=${UID},gid=${GID} \
-    rm ~/.julia && \
-    mkdir ~/.julia && \
-    cp -rp ${JULIA_DEPOT_CACHE}/* ~/.julia && \
-    julia -e 'using Pkg, Dates; Pkg.gc(collect_delay=Day(0))'
+RUN --mount=type=cache,target=${JULIA_USER_DEPOT_CACHE},uid=${UID},gid=${GID} \
+    rm ${JULIA_USER_DEPOT} && \
+    mkdir ${JULIA_USER_DEPOT} && \
+    cp -rp ${JULIA_USER_DEPOT_CACHE}/* ${JULIA_USER_DEPOT} && \
+    JULIA_LOAD_PATH=":" julia -e 'using Pkg, Dates; Pkg.gc(collect_delay=Day(0))'
 
 #####
-##### ray-jl stage
+##### build-ray-jl stage
 #####
 
-FROM ray-base as ray-jl
+FROM ray-base as build-ray-jl
 
 # Install Bazel and compilers
 ARG BAZEL_CACHE=/mnt/bazel-cache
-RUN set -eux; \
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
```

Comment:
why the additiona of new cache targets?

Reply:
This is just an optimization and slightly improves performance when running these steps. Based upon https://docs.docker.com/engine/reference/builder/#example-cache-apt-packages. Looks like I missed a setting as part of this which I'll add in now